Update token-cache-serialization.md (#249)
MicrosoftDocs/microsoft-authentication-library-dotnet: msal-dotnet-articles/how-to/token-cache-serialization.md

|  `TokenSource` | Source of the token. Tokens are retrieved from the cache much faster (for example, ~100 ms versus ~700 ms). Can be used to monitor and alarm the cache hit ratio. | Use with `DurationTotalInMs`. |
|  `CacheRefreshReason` | Reason for fetching the access token from the identity provider. | Use with `TokenSource`. |

## Size approximations

When using a token cache, it's important to consider the potential size of the cache, especially for highly-available and distributed applications. The current estimates are:

> ### Application

Reviewer: ```suggestion
### Application flows
```

> - Access tokens are about 2-3KB; 1 token per _app client ID_ \* number of tenants \* downstream resources.

Reviewer: ```suggestion
- Only access tokens are cached. One token about 2-3KB when persisted. There will be 1 token per _app client ID_ \* tenants \* downstream resources. For example a multi-tenanted app serving 1000 tenants and needing tokens for Graph and SharePoint will use: 3KB * 1000 * 2 i.e. approximately 6 MB.
```

> ### User

Reviewer: ```suggestion
### User Flows 
```

- **Access tokens** – 4KB; 1 token per _app client ID_ \* user \* tenant \* downstream resource.
- **Refresh token** – 2KB; 1 token per _client app ID_ \* user.
- **ID token** – 2KB; 1 token per _client app ID_ \* user \* number of tenants where that user logs in.

> 

Reviewer: ### Web API calling other web api (AcquireTokenOnBehalfOf)

Note: we strongly recommend using the higher level APIs from [Microsoft.Identity.Web](https://github.com/AzureAD/microsoft-identity-web/) for this and not MSAL directly. The caching considerations are the same.

Same as for web site scenario, but there will be 1 node for each session, not for each user. By default, MSAL identifies a session by hashing the upstream assertion, but this can be changed. See [Long Running OBO Processes.](https://learn.microsoft.com/en-us/entra/msal/dotnet/acquiring-tokens/web-apps-apis/on-behalf-of-flow#long-running-obo-processes)
